### PR TITLE
Add Bayesian Normal Density estimator

### PR DESCRIPTION
**Pull request by mamikonyana**

### CODE REVIEW

**Comment by mamikonyana** [comment]
If you want theoretical background, this is a good overview of what's happening in 1 dimensional case (section 3) - http://www.cs.berkeley.edu/~jordan/courses/260-spring10/lectures/lecture5.pdf

This are just results for multivariate case - http://www.fil.ion.ucl.ac.uk/~wpenny/publications/bmn.pdf

**Comment by pbailis** [comment]
So, just to check my understanding: this fits a single Normal distribution using Bayesian inference?

**Comment by mamikonyana** [comment]
Yes that's correct

File: src/main/java/macrobase/analysis/stats/BayesianNormalDensity.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import org.apache.commons.math3.linear.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+
+public class BayesianNormalDensity extends BatchTrainScore {
```
Comment (pbailis) [inline src/main/java/macrobase/analysis/stats/BayesianNormalDensity.java]: Comment explaining class would be helpful.

File: src/main/java/macrobase/analysis/stats/distribution/MultivariateTDistribution.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+        double determinant = new LUDecomposition(covarianceMatrix).getDeterminant();
+
+        this.multiplier = halfGammaRatio(dimensions + degreesOfFreedom, degreesOfFreedom)/
+                Math.pow(Math.PI * degreesOfFreedom, 0.5 * dimensions) /
+                Math.pow(determinant, 0.5);
+    }
+
+
+    public double density(RealVector vector) {
```
Comment (pbailis) [inline src/main/java/macrobase/analysis/stats/distribution/MultivariateTDistribution.java]: Do we want to have a more generic distribution class?

**Comment by pbailis** [comment]
Minor comments; otherwise, LGTM to merge.

File: src/main/java/macrobase/analysis/stats/distribution/MultivariateTDistribution.java (hunk 2 above)
Comment (mamikonyana) [inline src/main/java/macrobase/analysis/stats/distribution/MultivariateTDistribution.java]: They have too many methods in the abstract class from apache, so let's keep it simple for now.

File: src/main/java/macrobase/analysis/stats/BayesianNormalDensity.java (hunk 1 above)
Comment (mamikonyana) [inline src/main/java/macrobase/analysis/stats/BayesianNormalDensity.java]: adding

**Pull request merged**